[ChuanXinNg] ip

ChuanXinNg opened the pull request:
## Roo
> You can only manage time if you track it right. - [Spica’s team](https://www.spica.com/blog/the-best-time-management-quotes)

Roo helps to track your tasks so that they stay organized. It's
- text-based
- personal
- ~~Easy~~ **_SUPER_ EASY** to use

All you need to do is just
1. download it from [here](https://github.com/ChuanXinNg/ip.git)
2. open it
3. add your tasks
4. leave your tasks for Roo to manage✨

The most important thing is... it is **FREE!!!**

Features:
- [x] Manage todo tasks
- [x] Manage deadlines and events
- [ ] Find a specific event
- [ ] Filter your tasks by date

You can use Roo to practice Java too. Here is the `main` method:
```java
    public static void main(String[] args) {
        Roo roo = new Roo("roo.txt");
        roo.run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

ChuanXinNg committed:
Rename, Greet, Exit

ChuanXinNg committed:
Echo

ChuanXinNg committed:
Add, List

ChuanXinNg committed:
Mark not done

ChuanXinNg committed:
Mark as done

ChuanXinNg committed:
ToDos, Events, Deadlines

ChuanXinNg committed:
Handle Errors

ChuanXinNg committed:
Delete

ChuanXinNg committed:
Delete

ChuanXinNg committed:
Automated Text UI Testing

ChuanXinNg committed:
Use Enums

ChuanXinNg committed:
load data when roo starts up

ChuanXinNg committed:
save

ChuanXinNg committed:
Merge branch 'branch-Level-7'

ChuanXinNg committed:
Dates and Times

ChuanXinNg committed:
Merge branch 'branch-Level-8'

ChuanXinNg committed:
improve dates and times

ChuanXinNg committed:
Use more OOP

ChuanXinNg committed:
Organize into packages

ChuanXinNg committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ChuanXinNg committed:
Use Gradle

ChuanXinNg committed:
Add JUnit Tests

ChuanXinNg committed:
Create a JAR file

ChuanXinNg committed:
Done JavaDoc, Follow the Coding Standard, Find

ChuanXinNg committed:
Coding Standard

ChuanXinNg committed:
Coding Standard Switch Cases

ChuanXinNg committed:
Coding Standard Final

ChuanXinNg committed:
JavaDoc

ChuanXinNg committed:
Follow the coding standard

ChuanXinNg committed:
Find

ChuanXinNg committed:
Merged branch-Level-9

ChuanXinNg committed:
Merged branch-A-JavaDoc

ChuanXinNg committed:
Merged branch-A-CodingStandard

zhanyang01 commented on src/main/java/roo/Roo.java:
```diff
@@ -0,0 +1,152 @@
+package roo;
+
+import java.time.LocalDate;
```
I like how imports are done well, imported classes are listed out explicitly, Good job!

zhanyang01 commented on src/main/java/roo/task/Task.java:
```diff
@@ -0,0 +1,78 @@
+     * Checks if the task is marked as done.
+     * @return true if the task is marked as done, false otherwise.
+     */
+    public boolean done() {
```
This should sound like a boolean, perhaps isItFinished will be better?

zhanyang01 commented on src/main/java/roo/Roo.java:
```diff
@@ -0,0 +1,152 @@
+/**
+ * The main class for the Roo application, which is a tasks management program.
+ */
+public class Roo {
```
Class names are all in PascalCase, variables and methods are in camelCase which is good! Keep up the good work!

zhanyang01 reviewed (commented):
I like how the code standards are met in terms of readability, Keep up the good job in the IP so far!

ChuanXinNg committed:
Deleted unused files

ChuanXinNg committed:
Done JavaFX tutorial

ChuanXinNg committed:
GUI v1

ChuanXinNg committed:
GUI v2

ChuanXinNg committed:
GUI v3

ChuanXinNg committed:
Use Varargs

ChuanXinNg committed:
merged branch

ChuanXinNg committed:
Debugging 1

ChuanXinNg committed:
Updated Roo initialise

ChuanXinNg committed:
More OOP Commands

ChuanXinNg committed:
More OOP Command

ChuanXinNg committed:
Add assertions to improve code robustness

This commit adds assertions to the Roo class so that potential issues
can be catch earlier during program execution.

This is because some input is assumed to be not null and assertions
help to check for potential mistakes that may accidentally set them
to null.

Let's add assertions to check that the "filepath" argument in the
constructor and the "input" argument in the "run" method is not null.

These changes will help in identifying and debugging issues more easily
during development.

Refer to this website on dealing with assertions
https://www.geeksforgeeks.org/assertions-in-java/

ChuanXinNg committed:
Deleted unused files

ChuanXinNg committed:
Improve Code Quality

ChuanXinNg committed:
Use Stream

ChuanXinNg committed:
Minor issues in Ui delete

ChuanXinNg committed:
Merge pull request #1 from ChuanXinNg/branch-A-Assertions

Add assertions to improve code robustness

ChuanXinNg committed:
Merge branch 'master' into branch-A-Streams

ChuanXinNg committed:
Merge pull request #2 from ChuanXinNg/branch-A-Streams

Use Stream

ChuanXinNg committed:
Merge branch 'master' into branch-A-CodeQuality

ChuanXinNg committed:
Merge pull request #3 from ChuanXinNg/branch-A-CodeQuality

Improve Code Quality

ChuanXinNg committed:
Add tag

ChuanXinNg committed:
Tag

ChuanXinNg committed:
Merge branch 'branch-C-Tagging'

ChuanXinNg committed:
untag

ChuanXinNg committed:
solved deadline and event issues when loading from storage

ChuanXinNg committed:
Better GUI

ChuanXinNg committed:
Merge branch 'branch-A-BetterGui'

ChuanXinNg committed:
Done

ChuanXinNg committed:
Final version

ChuanXinNg committed:
Update README.md

ChuanXinNg committed:
Update README.md

ChuanXinNg committed:
FINAL

ChuanXinNg committed:
Merged remote and local

ChuanXinNg committed:
tiny changes

ChuanXinNg committed:
corrected file path to images

ChuanXinNg committed:
javadocs changes
